src/encoded/static/components/experiment_series.js

```diff
@@ -543,26 +562,19 @@ class ExperimentSeriesComponent extends React.Component {
 
             // Experiment table data
             Object.keys(viewableDatasets).forEach((datasetAtId) => {
-                // Select one analysis based on file assembly and file count
-                // for extracting quality metrics
-                const filesByDesiredAssembly = [];
-                viewableDatasets[datasetAtId].files.forEach((f) => {
-                    if (selectedAssembly.includes(f.assembly)) {
-                        filesByDesiredAssembly.push(f['@id']);
-                    }
-                });
+                // Select one analysis based on analysis award
                 const analysisObjects = viewableDatasets[datasetAtId].analysis_objects || [];
-                const selectedAnalysis = analysisObjects.filter(
+                viewableDatasets[datasetAtId].selectedAnalysis = analysisObjects.filter(
                     (analysis) => selectedAssembly.includes(analysis.assembly)
-                        && analysis.pipeline_award_rfas.includes(selectedAnalysisRFA)
+                        && analysis.pipeline_award_rfas.includes(this.state.selectedAnalysisRFA)
                         && analysis.pipeline_labs.includes(selectedAnalysisLab)
                 )[0] || { files: [] };
-                viewableDatasets[datasetAtId].files = viewableDatasets[datasetAtId].files.filter(
-                    (f) => selectedAnalysis.files.includes(f['@id']),
+                viewableDatasets[datasetAtId].selectedFiles = viewableDatasets[datasetAtId].files.filter(
+                    (f) => viewableDatasets[datasetAtId].selectedAnalysis.files.includes(f['@id'])
                 );
                 // Number of subrows to be shown corresponds to the number of biological replicates. Show at least one by default.
                 let bioRepNums = _.uniq(
-                    viewableDatasets[datasetAtId].files.map((f) => f.biological_replicates).reduce(
+                    viewableDatasets[datasetAtId].selectedFiles.map((f) => f.biological_replicates).reduce(
                         (bioRepArray, reps) => bioRepArray.concat(reps), []
                     )
```

Comment:
Does this `reduce` do anything? I feel like it just outputs a copy of what was given from the output of `map`.

Reply:
You are correct. I should just reduce as you suggested. Fixed.